src/modules/auth/controller.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const { validateAuth } = require('./validators');
+const validator = require('../../middlewares/validator');
+const passwordService = require('../password/service');
+const userService = require('../user/service');
+const { INVALID_CREDENTIALS_MESSAGE, TOKEN_EXPIRATION_TIME } = require('./constants');
+const { StatusCodes } = require('http-status-codes');
+const { UnauthorizedError } = require('../../utils/errors');
+const authService = require('./service');
+
+const router = express.Router();
+
+const tryAuth = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const user = await userService.getUser(email);
+    if (user) {
+      await passwordService.compare(password, user.password);
```

Comment:
So in combination of those changes in `compare` func you dont need extra condition for user
You can get user object and pass user.password in compare func like `.compare(password, user && user.password)`